src/telemetry/activator.ts

```diff
@@ -19,41 +19,64 @@
 
 import { BallerinaExtension, ExtendedLangClient } from "src/core";
 import { window } from "vscode";
-import { getTelemetryProperties, TM_ERROR_LANG_SERVER, TM_EVENT_LANG_SERVER } from ".";
+import { getTelemetryProperties, TM_ERROR_LANG_SERVER, TM_EVENT_LANG_SERVER, TM_FEATURE_USAGE_LANG_SERVER } from ".";
+import { TM_EVENT_TYPE_ERROR, TM_EVENT_TYPE_FEATURE_USAGE } from "./events";
 
 export function activate(ballerinaExtInstance: BallerinaExtension) {
     const reporter = ballerinaExtInstance.telemetryReporter;
     const langClient = <ExtendedLangClient>ballerinaExtInstance.langClient;
 
     // Start listening telemtry events from language server
-    ballerinaExtInstance.onReady().then(() => {
-        langClient.onNotification('telemetry/event', (event: LSErrorTelemetryEvent | LSTelemetryEvent) => {
-            if ('errorMessage' in event) {
-                const errorEvent: LSErrorTelemetryEvent = event;
-                const props: { [key: string]: string; } = getTelemetryProperties(ballerinaExtInstance, event.component, event.message)
-                props["ballerina.langserver.error.stacktrace"] = errorEvent.errorStackTrace;
-                props["ballerina.langserver.error.message"] = errorEvent.errorMessage;
-                reporter.sendTelemetryEvent(TM_ERROR_LANG_SERVER, props);
-            } else {
-                reporter.sendTelemetryEvent(TM_EVENT_LANG_SERVER, getTelemetryProperties(ballerinaExtInstance, event.component, event.message));
-            }
-        });
-    })
+    ballerinaExtInstance.onReady()
+        .then(() => {
+            langClient.onNotification('telemetry/event', (event: LSTelemetryEvent) => {
+                let props: { [key: string]: string; };
+                switch (event.type) {
+                    case TM_EVENT_TYPE_ERROR:
+                        const errorEvent: LSErrorTelemetryEvent = <LSErrorTelemetryEvent>event;
+                        props = getTelemetryProperties(ballerinaExtInstance, event.component, errorEvent.message);
+                        props["ballerina.langserver.error.description"] = errorEvent.message;
```

Comment:
Shall we avoid duplicate properties? `ballerina.message` & `ballerina.langserver.error.description` has the same value.

Reply:
I did that for the completion of the data that we will be querying. When examining the data, having everything related to the `error` with the prefix, `ballerina.langserver.error` makes more sense and avoids the requirement of looking at the plugin source code to identify which field is which. Same has been done for `ballerina.langserver.feature.name`. WDYT?

Reply:
As discussed offline, tried to pass `null` as the 3rd parameter, but gave an error because the `getTelemetryProperties()` method wants a string as the 3rd parameter. Therefore, decided to pass in the event type constant. WDYT? @prabushi